pkgs/development/compilers/vyper/default.nix

```diff
@@ -10,8 +10,18 @@ let
     def __init__(foo: address):
         self.count = 1
   '';
-in
 
+  # Vyper requires semantic-version on v2.8.5
+  semantic-version-285 = semantic-version.overridePythonAttrs (old: rec {
+    pname = "semantic_version";
+    version = "2.8.5";
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "d2cb2de0558762934679b9a104e82eca7af448c9f4974d1f3eeccff651df8a54";
+    };
+  });
```

Comment:
Ok, this confused me a lot because this is outside the `python-modules` but it is a Python package.

We can't really do this kinda of override. The issue is because if we try to put at the same `buildInputs` both this package and another package that depends on `semantic-version`, weird things will happen. If this was outside the `python-modules` (as a Python application instead) it would be fine because you can't really add it to `PYTHONPATH`, but the way it is currently is is dangerous.

Reply:
What would be the recommended action in this case?

Reply:
The solution here instead is to relax this dependency, basically converting it from `semantic-version==2.8.5` to `semantic-version`. This used to be done manually in the past (using `sed` or `substituteInPlace`), but lucky for you https://github.com/NixOS/nixpkgs/pull/170813 is already available in `master` so you can do this instead:

```nix
{ ..., pythonRelaxDepsHook, ... }: # add it to inputs

{
  ...
  nativeBuildInputs = [ pythonRelaxDepsHook ];
  pythonRelaxDeps = [ "semantic-version" ];
}
```

Reply:
>  I don't feel like modifying the requirements through a replacement to match the current version of semantic-tools would be recommended if I don't know how it works (can be prone to breaking the package).

Sadly this is the best solution we have. We can't afford to have two different Python dependencies with the same name inside a Python environment.

Also, this bump looks like very minor. It is unlikely to cause any issues.

Reply:
Also, it looks like it is a build dependency (https://pypi.org/project/semantic-version/)? So it will either works or it will not work.